ratbagd/ratbagd-device.c

```diff
@@ -385,7 +385,8 @@ int ratbagd_device_new(struct ratbagd_device **out,
 
 	device->ctx = ctx;
 	rbnode_init(&device->node);
-	device->lib_device = ratbag_device_ref(lib_device);
+	/* note: we steal the reference here */
+	device->lib_device = lib_device;
 
```

Comment:
no, sorry. never steal references, the whole point of having ref/unref is that you don't have to worry about who has the pointer, you always get a reference and unref it when you're done with it.

this can be fixed properly with the goto error: approach that calls unref if device->lib_device is not NULL